### DISCUSSION TITLE
Implement a Record object

### DISCUSSION THREAD

**Discussion by syniex** [oracle/python-oracledb#301] [Ideas]
Currently, when retrieving results from a database, many of us construct dictionaries from lists of tuples to provide a more convenient API for working with the data. This approach is common but might not be the most efficient.

We could create a Record class in Cython that encapsulates the data returned from the database. This class would allow for easier access to column values and could potentially offer performance benefits over the current approach of using dictionaries.

[asyncpg Record API Documentation](https://magicstack.github.io/asyncpg/current/api/index.html#record-objects)

**Comment by anthony-tuininga** [comment]
I moved this to an enhancement request (#302)
